src/seed_tools/utils/seed_validation.ts

```diff
@@ -0,0 +1,260 @@
+export function validateSeedReturnErrors(seed: VariationsSeed): string[] {
+  return checkOverlappingStudies(seed);
+}
+
+// Validates a seed by checking for overlapping studies that use the same
+// feature with different filters.
+function checkOverlappingStudies(seed: VariationsSeed): string[] {
+  const errors: string[] = [];
+  // Create a map from feature name to studies that use that feature.
+  const featureNamesToStudies = new DefaultMap<string, Study[]>(() => []);
+  for (const study of seed.study) {
+    const usedFeatureNames = new Set<string>();
+    for (const experiment of study.experiment) {
+      const featureAssociation = experiment.feature_association;
+      if (featureAssociation !== undefined) {
+        for (const enableFeature of featureAssociation.enable_feature) {
+          usedFeatureNames.add(enableFeature);
+        }
+        for (const disableFeature of featureAssociation.disable_feature) {
+          usedFeatureNames.add(disableFeature);
+        }
+      }
+    }
+
+    for (const usedFeatureName of usedFeatureNames) {
+      featureNamesToStudies.get(usedFeatureName).push(study);
+    }
+  }
+
+  // Date range cache for each study.
+  const studyDateRanges = new DefaultMap<Study, study_filter_utils.DateRange>(
+    study_filter_utils.getStudyDateRange,
+  );
+
+  // Version range cache for each study.
+  const studyVersionRanges = new DefaultMap<
+    Study,
+    study_filter_utils.VersionRange
+  >(study_filter_utils.getStudyVersionRange);
+
+  // OS version range cache for each study.
+  const studyOsVersionRanges = new DefaultMap<
+    Study,
+    study_filter_utils.VersionRange
+  >(study_filter_utils.getStudyOsVersionRange);
+
+  const filterIntersectCheckers = createStudyFilterIntersectCheckers({
+    start_date: (study1: Study, study2: Study) =>
+      doesDateRangeIntersect(
+        studyDateRanges.get(study1),
+        studyDateRanges.get(study2),
+      ),
+    end_date: () => true, // Handled by start_date.
+    min_version: (study1: Study, study2: Study) =>
+      doesVersionRangeIntersect(
+        studyVersionRanges.get(study1),
+        studyVersionRanges.get(study2),
+      ),
+    max_version: () => true, // Handled by min_version.
+    min_os_version: (study1: Study, study2: Study) =>
+      doesVersionRangeIntersect(
+        studyOsVersionRanges.get(study1),
+        studyOsVersionRanges.get(study2),
+      ),
+    max_os_version: () => true, // Handled by min_os_version.
+    channel: (study1: Study, study2: Study) =>
+      doesFilterArrayIntersect(study1.filter?.channel, study2.filter?.channel),
+    platform: (study1: Study, study2: Study) =>
+      doesFilterArrayIntersect(
+        study1.filter?.platform,
+        study2.filter?.platform,
+      ),
+    locale: (study1: Study, study2: Study) =>
+      doesFilterArrayIntersect(study1.filter?.locale, study2.filter?.locale),
+    exclude_locale: (study1: Study, study2: Study) =>
+      doesFilterArrayIntersect(
```

Comment:
I'm not sure about `exclude_*` logic..
If we have study A with `exclude_locale: ["en"]` and study B with `exclude_locale: ["es"]`, they actually intersect (e.g. if user locale is `fr`) .